server/gorillamux/handlers.go

```diff
@@ -0,0 +1,42 @@
+package gorillamux
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+
+	"github.com/SKF/go-rest-utility/problems"
+	handlerproblems "github.com/SKF/go-rest-utility/server/gorillamux/problems"
+)
+
+func MethodNotFoundHandler(router mux.Router) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		allowedMethods := []string{}
+
+		router.Walk(func(route *mux.Route, router *mux.Router, ancestors []*mux.Route) error { //nolint: errcheck
```

Comment:
Should we consider a case where no methods were found?

Reply:
Then you don't get to this handler?

Reply:
Maybe the route.GetMethods call failed? But yes we would assume at least one route should match for this handler to be called.

Reply:
If you have set up an route with no methods that can happen. Or I guess it depends how much going to mess with the matchers but I dunno seems highly unlikely getting that issue after we already matched.

Reply:
Agree